kevm-pyk/src/kontrol/foundry.py

```diff
@@ -124,14 +125,14 @@ def digest(self) -> str:
 
     @cached_property
     def llvm_dylib(self) -> Path | None:
-        from kevm_pyk.kompile import Kernel
-
-        arch = Kernel.get()
         foundry_llvm_dir = self.out / 'kompiled-llvm'
-        if arch == Kernel.LINUX:
-            dylib = foundry_llvm_dir / 'interpreter.so'
-        else:
-            dylib = foundry_llvm_dir / 'interpreter.dylib'
+        match sys.platform:
+            case 'linux':
+                dylib = foundry_llvm_dir / 'interpreter.so'
+            case 'darwin':
+                dylib = foundry_llvm_dir / 'interpreter.dylib'
+            case _:
+                raise ValueError('Unsupported platform: {sys.platform}')
```

Comment:
There is a logic change here to stick with https://github.com/runtimeverification/pyk/pull/564/files#diff-30de5e61b136df5b085f8a5807eab3560667b41760f0eb0a0ad21220299fb3efR788-R795